be/src/services/vocabulary.service.ts

```diff
@@ -50,6 +50,55 @@ export class VocabularyService {
             });
     }
 
+    public updateVocab(id: string, ifUnmodifiedSince: Date, newVocab: Vocabulary): Promise<Vocabulary> {
+        // @ts-ignore
+        return this.getVocabular(id).then((dbo: Vocabulary) => {
+            if (!!dbo) {
+                if (dbo.lastModified.getTime() <= ifUnmodifiedSince.getTime()) {
+                    const filter: Filter<Vocabulary> = {
+                        _id: new ObjectId(id),
+                        lastModified: {
+                            // eslint-disable-rows-line @typescript-eslint/no-unsafe-argument
+                            $eq: ifUnmodifiedSince
+                        }
+                    };
+
+                    const update: UpdateFilter<Vocabulary> = {
+                        $set: {
+                            _id: new ObjectId(id),
+                            label: newVocab.label,
+                            description: newVocab.description
+                        },
+                        $currentDate: {
+                            lastModified: true
+                        }
+                    };
+
+                    // @ts-ignore
+                    return this.collection.updateOne(filter, update, {upsert: false}).then((result: UpdateResult) => {
```

Comment:
Using `{ new: true }` will return you the updated/modified version from the database.

See https://docs.mongodb.com/manual/reference/method/db.collection.findAndModify/

Reply:
For the protocol, the correct link for our mongoDB is
https://docs.mongodb.com/manual/reference/method/db.collection.findOneAndUpdate/